core/src/saros/git/JGitFacade.java

```diff
@@ -0,0 +1,190 @@
+package saros.git;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.log4j.Logger;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.MergeCommand.FastForwardMode;
+import org.eclipse.jgit.api.RemoteRemoveCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.errors.RevisionSyntaxException;
+import org.eclipse.jgit.lib.NullProgressMonitor;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.transport.BundleWriter;
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.TransportBundleStream;
+import org.eclipse.jgit.transport.URIish;
+
+public class JGitFacade {
+  private static final Logger log = Logger.getLogger(JGitFacade.class);
+  private File workDirTree;
+
+  public JGitFacade(File workDirTree) {
+    setWorkDirTree(workDirTree);
+  }
+
+  /**
+   * Create a {@code bundle} with all commits from {@code basis} to {@code actual} of the Git repo.
+   *
+   * @param actual The name of the ref to lookup. For example: "HEAD" or "refs/heads/master". Must
+   *     not be a short-handform; e.g., "master" is not automatically expanded
+   *     to"refs/heads/master". Include the object the ref is point at in the {@code bundle} and (if
+   *     the {@code basis} is an empty string) everything reachable from it.
+   * @param basis Assume that the recipient have at least the commit the {@code basis} is pointing
+   *     to. In order to fetch from a {@code bundle} the recipient must have the commit the {@code
+   *     basis} is pointing to.
+   * @throws IllegalArgumentException {@code workDirTree} is null or if resolving {@code actual}
+   *     leads to a not existing ref
+   * @throws IOException failed while read/resolved parameters or while written to the {@code
+   *     bundle}
+   */
+  public byte[] createBundle(String actual, String basis)
+      throws IllegalArgumentException, IOException {
+
+    Git git = getGit();
```

Comment:
You call getGit() here in line 108, 139, 150, 173. Why don't you create a private final variable which is instantiated in the constructor  and reused in all methods?

Reply:
done